src/DaPigGuy/libPiggyEconomy/providers/EconomyProvider.php

```diff
@@ -22,11 +22,31 @@ public function getMonetaryUnit(): string
         return "$";
     }
 
-    abstract function getMoney(Player $player): float;
-
-    abstract function giveMoney(Player $player, float $amount): void;
-
-    abstract function takeMoney(Player $player, float $amount): void;
-
-    abstract function setMoney(Player $player, float $amount): void;
+	/**
+	 * $callback -> function(?float $money): void{}
+	 *
+	 * The $money will be null if player is not found.
+	 */
+    abstract function getMoney(Player $player, callable $callback): void;
+
+	/**
+	 * $callback -> function(bool $success): void{}
+	 *
+	 * If money was given successfully then callable would contain true else false.
+	 */
+    abstract function giveMoney(Player $player, float $amount, ?callable $callback = null): void;
+
+	/**
+	 * $callback -> function(bool $success): void{}
+	 *
+	 * If money was taken successfully then callable would contain true else false.
+	 */
+    abstract function takeMoney(Player $player, float $amount, ?callable $callback = null): void;
+
+	/**
+	 * $callback -> function(bool $success): void{}
+	 *
+	 * If money was set successfully then callable would contain true else false.
+	 */
+    abstract function setMoney(Player $player, float $amount, ?callable $callback = null): void;
```

Comment:
@DaPigGuy dude get this he explains his code lol unlike someone

Reply:
anyways need to reformat

Reply:
kk